pkg/worker/worker.go

```diff
@@ -91,7 +97,6 @@ func sigHandler() <-chan struct{} {
 		signal.Notify(sigc, syscall.SIGTERM)
 		sig := <-sigc
 		logrus.WithField("signal", sig).Info("got a signal, waiting then sending the real shutdown signal")
-		close(stop)
```

Comment:
channels in a select should not be closed because they are always ready and don't give other channels a chance to get read from.